core/include/drivers/clk.h

```diff
@@ -6,60 +6,116 @@
 #ifndef __DRIVERS_CLK_H
 #define __DRIVERS_CLK_H
 
+#include <assert.h>
 #include <kernel/refcount.h>
 #include <stdint.h>
 #include <tee_api_types.h>
+#include <util.h>
 
 /* Flags for clock */
 #define CLK_SET_RATE_GATE	BIT(0) /* must be gated across rate change */
 #define CLK_SET_PARENT_GATE	BIT(1) /* must be gated across re-parent */
 
+/*
+ * Type of clock instantiated:
+ *
+ * CLK_OPS_ELT identities a full fledged clock. The struct clk * reference can be
```

Comment:
I'm really not sure "element" is the good word here, but I can't come up with something more meaningful...
child ? hierarchical ? tree_element ?

Reply:
`CLK_OPS_GENERIC` / `is_clk_gen()` ?
`CLK_OPS_STANDARD` / `is_clk_std()` ?
`CLK_OPS_FULLFEATURED` / `is_clk_ff()` ?

Reply:
`CLK_OPS_FULLFEATURED` /` is_clk_ff()` looks good to me ! Did not wanted it at first because it's too long but `ff`/`lw` is readable IMHO.